packages/tailwindcss/src/compat/screens-config.test.ts

```diff
@@ -0,0 +1,592 @@
+        @media (width < 48rem) {
+          text-decoration-line: underline;
+        }
+      }
+    }
+    .min-sm\\:max-md\\:underline {
+      @media (width >= 40rem) {
+        @media (width < 48rem) {
+          text-decoration-line: underline;
+        }
+      }
+    }
+    .md\\:flex {
+      @media (width >= 48rem) {
+        display: flex;
+      }
+    }
+    .midi\\:flex {
+      @media (width >= 48rem) {
+        display: flex;
+      }
+    }
+    .min-md\\:max-lg\\:underline {
+      @media (width >= 48rem) {
+        @media (width < 64rem) {
+          text-decoration-line: underline;
+        }
+      }
+    }
+    .min-midi\\:max-maxi\\:underline {
+      @media (width >= 48rem) {
+        @media (width < 64rem) {
+          text-decoration-line: underline;
+        }
+      }
+    }
+    .maxi\\:flex {
+      @media (width >= 64rem) {
+        display: flex;
+      }
+    }
+    .print\\:items-end {
+      @media print {
+        align-items: flex-end;
+      }
+    }
+    "
+  `)
+})
+
+test('JS config with `theme: { extends }` should not include the `default-config` values', async () => {
+  let input = css`
+    @config "./config.js";
+    @tailwind utilities;
+  `
+
+  let compiler = await compile(input, {
+    loadConfig: async () => ({
+      theme: {
+        extend: {
+          screens: {
+            mini: '40rem',
+            midi: '48rem',
+            maxi: '64rem',
+          },
+        },
+      },
+    }),
+  })
+
+  expect(
+    compiler.build([
+      'sm:flex',
+      'md:flex',
+      'mini:flex',
+      'midi:flex',
+      'maxi:flex',
+      'min-md:max-lg:underline',
+      'min-sm:max-md:underline',
+      'min-midi:max-maxi:underline',
+      'min-mini:max-midi:underline',
+
+      // Ensure other core variants appear at the end
+      'print:items-end',
+    ]),
+  ).toMatchInlineSnapshot(`
+    ".mini\\:flex {
+      @media (width >= 40rem) {
+        display: flex;
+      }
+    }
+    .min-mini\\:max-midi\\:underline {
+      @media (width >= 40rem) {
+        @media (width < 48rem) {
+          text-decoration-line: underline;
+        }
+      }
+    }
+    .midi\\:flex {
+      @media (width >= 48rem) {
+        display: flex;
+      }
+    }
+    .min-midi\\:max-maxi\\:underline {
+      @media (width >= 48rem) {
+        @media (width < 64rem) {
+          text-decoration-line: underline;
+        }
+      }
+    }
+    .maxi\\:flex {
+      @media (width >= 64rem) {
+        display: flex;
+      }
+    }
+    .print\\:items-end {
+      @media print {
+        align-items: flex-end;
+      }
+    }
+    "
+  `)
+})
+
+describe('complex screen configs', () => {
+  test('generates utilities', async () => {
+    let input = css`
+      @config "./config.js";
+      @tailwind utilities;
+    `
+
+    let compiler = await compile(input, {
+      loadConfig: async () => ({
+        theme: {
+          extend: {
+            screens: {
+              sm: { max: '639px' },
```

Comment:
I feel like, if we can, we should make `max-sm` and `sm` the same when given a config like this.

Reply:
That’s not how v3 works though iirc so might not make sense to add support for it in v4?

Reply:
ah yeah right, okay then.